sklearn/utils/validation.py

```diff
@@ -802,6 +802,8 @@ def check_array(
     # DataFrame), and store them. If not, store None.
     dtypes_orig = None
     pandas_requires_conversion = False
+    # track if we have a Series-like object to raise a better error message
+    type_if_series = None
```

Comment:
Nit: I think it's a little cleaner to use `bool`:

```suggestion
    is_series_like = False
```

Reply:
@thomasjpfan Well, the reason I'm not using a `bool` here is I mentioned the explicit type of the object in the error message, but the type of the object has been converted at the point we raise the error message, so I chose to keep track of the type in the first place. If you think that we do not need to explicitly mention the type but only say Series-like, then I can use `bool` here.

Reply:
Ah I see. Okay, lets go with keeping the type around.